src/core/credentials/transport/tls/spiffe_utils.h

```diff
@@ -79,12 +86,21 @@ class SpiffeBundle final {
     return kLoader;
   }
 
+  // Needed to handle the STACK_OF(X509)* root stack
+  SpiffeBundle() = default;
+  ~SpiffeBundle();
+  SpiffeBundle(const SpiffeBundle& other);
+  SpiffeBundle& operator=(const SpiffeBundle& other);
+
   void JsonPostLoad(const Json& json, const JsonArgs&,
                     ValidationErrors* errors);
 
   // Returns a vector of the roots in this SPIFFE Bundle.
   absl::Span<const std::string> GetRoots();
 
+  // The caller does not take ownership of the stack of roots.
+  STACK_OF(X509) * GetRootStack() { return *root_stack_; }
+
   bool operator==(const SpiffeBundle& other) const {
```

Comment:
Consider adding a comment in the equality check that it is sufficient to compare the PEM-encoded roots.